File: app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Models/LocationObject.java
package ithaca_transit.android.cornellappdev.com.ithaca_transit.Models;

import com.google.android.gms.maps.model.LatLng;

public class LocationObject {
    public static final double BLANK_COORD = 0.0D;

    private Double latitude;
    private Double longitude;
    private String name;

    public LocationObject(String name, String id, double latitude, double longitude) {
        super();

        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public LocationObject() {
        this("", "", BLANK_COORD, BLANK_COORD);
    }

    public LocationObject(String name, String id) {
        this(name, id, BLANK_COORD, BLANK_COORD);
    }

    public LocationObject(Double latitude, Double longitude) {
        this("", "", latitude, longitude);
    }

    public LocationObject(String name, double latitude, double longitude) {
        this(name, "", latitude, longitude);
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getLatitude() {
        return this.latitude;
    }

    public void setLatitude(Double var1) {
        this.latitude = var1;
    }

    public Double getLongitude() {
        return this.longitude;
    }

    public void setLongitude(Double var1) {
        this.longitude = var1;
    }

    public LatLng coordinates() {
        return new LatLng(latitude, longitude);
    }

}

File: app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Models/LocationObject.kt
package ithaca_transit.android.cornellappdev.com.ithaca_transit.Models

import com.google.android.gms.maps.model.LatLng

class LocationObject(name: String, id: String, latitude: Double, longitude: Double){
    var id: String ?= null
    var name: String ?= null
    var latitude: Double ?= null
    var longitude: Double ?= null

    init {
        this.id = id
        this.name = name
        this.latitude = latitude
        this.longitude = longitude
    }

    constructor(): this("", "", 0.0, 0.0)

    constructor(name: String): this(name, "", 0.0, 0.0)

    constructor(name: String, latitude: Double, longitude: Double):this(name,"",latitude,
            longitude)

    fun coordinates(): LatLng {
        return LatLng(latitude!!, longitude!!)
    }
}